ARCHITECTURE.md

```diff
@@ -68,16 +68,19 @@ the server understands ("SolidDatasets"), and those that do not ("Files").
 The main difference between the two is that, by virtue of the server
 understanding the structure of the former, we can do partial updates (i.e.
 PATCH requests) and retrieve the data in different formats (currently, the
-spec mandates that servers can serve such data as at least Turtle and JSON-LD).
+spec mandates that servers can receive/return such data as at least Turtle and JSON-LD).
 
-Thus, data stored as Turtle or JSON-LD can be fetched as a SolidDataset; the
+Thus, data requested as Turtle or JSON-LD can be fetched as a SolidDataset; the
```

Comment:
Well technically, you can request any representation for a Resource using the `Accept` header, and it's up to the server to decide whether your request makes sense or not. In that sense, what really matters is not as much how the data is stored, but indeed how it is requested which matters, because you could technically request some JPG file as RDF, and the server would either return a representation in the requested format, or politely tell you off :)

Reply:
Yeah, this is tricky and nuanced. Data is only ever `stored` in Solid as either binary data, or RDF data. The spec says that Solid servers must accept and deliver back RDF data via a minimum of two RDF serializations, Turtle and JSON-LD. Notice that previous sentence doesn't mention the word `store` - and that's because its completely up to the Server implementation to decide how it stored RDF data - i.e., even if I POST a payload of Turtle to the server, it's totally free to store that as N-Quads if it wants, or store it in a relational database (which is what ESS does today)). And anyone else is totally free to later request that same resource as either JSON-LD or Turtle, and the Server MUST honor that request and return that resource in that serialization.
But that same Server is also completely free to offer support for receiving and delivering that exact same resource back using any other RDF serializations too (e.g., Trig, N-Triples, Trix, RDF/XML, etc.), if a user wishes to request that of the Server, and the Server happens to offer support for that RDF serialization.
And as NicAS says, a user could indeed request that very same RDF resource back from the Server with an `Accept: image/jpeg` header - the answer to that 'crazy request' is up to the Server - it could reply with a "400 Bad Request", or a "406 Not Acceptable", or "415 Unsupported Media Type", or "[418 I'm a teapot](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/418)" - or indeed it might actually reply with a "200 Ok" and return an actual JPEG image of something that the Server deems is appropriate for that particular resource.
Hopefully that's a bit clearer :) !